### PR TITLE
ArC: remove deprecated classes and methods

### PR DESCRIPTION
**Pull request by mkouba**

### CODE REVIEW

**Comment by Ladicek** [comment]
`ProxyUnwrapperBuildItem` was deprecated in 3.0, so this is fairly soon IMHO. My personal opinion is that 6 months/releases is a reasonable period of time (which would be exactly the case for `ClientProxyUnwrapper` that got deprecated in 2.13). WDYT?

That's my only reservation about this, so if you think it's fine, I of course won't block it :-)

**Comment by mkouba** [comment]
> `ProxyUnwrapperBuildItem` was deprecated in 3.0, so this is fairly soon IMHO. My personal opinion is that 6 months/releases is a reasonable period of time (which would be exactly the case for `ClientProxyUnwrapper` that got deprecated in 2.13). WDYT?
> 
> That's my only reservation about this, so if you think it's fine, I of course won't block it :-)

I do want to remove as much deprecated stuff as possible for the 3.2 (aka LTS).

**Comment by mkouba** [comment]
`io.quarkus.arc.Priority` is another good candidate...

**Comment by manovotn** [comment]
> I do want to remove as much deprecated stuff as possible for the 3.2 (aka LTS).

I am fully behind this :+1: 
We did the deprecations for a reason and an LTS is a good cut-off point at which we should perform a removal so that we don't carry on with the legacy for far too long.

File: independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/alternatives/priority/AlternativePriorityResolutionTest.java [hunk 1]
```diff
@@ -45,7 +45,7 @@ static class NoParentAlternativePriorityProducer1 {
         @Produces
         @Singleton
         @Alternative
-        @io.quarkus.arc.Priority(4)
+        @Priority(4)
```
Comment (manovotn) [inline independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/alternatives/priority/AlternativePriorityResolutionTest.java:48]: This test is now likely superfluous + the TCKs we are now running are covering this as well.

File: independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/instance/PriorityTest.java [hunk 2]
```diff
@@ -35,8 +35,7 @@ interface Converter {
 
     }
 
-    @jakarta.annotation.Priority(5) // this priority takes precedence
-    @Priority(50)
+    @Priority(5)
```
Comment (manovotn) [inline independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/instance/PriorityTest.java:38]: This tested conflict of both annotations - not needed anymore

File: independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/stereotypes/StereotypeAlternativeArcPriorityTest.java [hunk 3]
```diff
@@ -19,7 +20,6 @@
 import org.junit.jupiter.api.extension.RegisterExtension;
 
 import io.quarkus.arc.Arc;
-import io.quarkus.arc.Priority;
 import io.quarkus.arc.test.ArcTestContainer;
 
 // copy of `StereotypeAlternativeTest` that uses ArC `@Priority`
```
Comment (manovotn) [inline independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/stereotypes/StereotypeAlternativeArcPriorityTest.java:25]: As per this comment, I don't think we need this test anymore :)

**Review by manovotn — COMMENTED** [review]
Just a few minor comments WRT to now superfluous tests

File: independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/alternatives/priority/AlternativePriorityResolutionTest.java (hunk 1 above)
Comment (manovotn) [reply independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/alternatives/priority/AlternativePriorityResolutionTest.java:48]: Actually, this test might overlap with some others but it still probably valuable, let's keep it in.

File: independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/instance/PriorityTest.java (hunk 2 above)
Comment (manovotn) [reply independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/instance/PriorityTest.java:38]: We could remove of the beans but we still want to keep what the test asserts (ordering of instances while iterating over them) so I've just kept this as-is.

**Review by manovotn — APPROVED** [review]

**Review by Ladicek — APPROVED** [review]
LGTM (barring the comment about perhaps removing some deprecations too quickly)

**Pull request merged**